wdl-grammar/src/experimental/grammar/v1.rs

```diff
@@ -890,3 +1144,419 @@ fn call_statement(parser: &mut Parser<'_>, _marker: Marker) -> Result<(), (Marke
+                .unwrap_or_else(|| (None, parser.span()));
+            return Err((
+                marker,
+                Error::Expected {
+                    expected: "expression",
+                    found,
+                    span,
+                    describe: Token::describe,
+                },
+            ));
+        }
+    };
+
+    // Extend the parent chain of the left-hand side to the provided marker.
+    lhs = lhs.extend_to(parser, marker);
+
+    loop {
+        // Check for either an infix or postix operation
+        match parser.peek() {
+            Some((token, _)) if INFIX_OPERATOR_EXPECTED_SET.contains(token.into_raw()) => {
+                // The operation is an infix operation; check the precedence level
+                let (precedence, kind, associativity) = infix_precedence(token);
+                if precedence < min_precedence {
+                    break;
+                }
+
+                let infix = lhs.precede(parser);
+                parser.next();
+
+                // Recuse for the right-hand side
+                let rhs = parser.start();
+                if let Err((rhs, e)) = expr_with_precedence(
+                    parser,
+                    rhs,
+                    // Add one to the precedence for left-associative operators
+                    match associativity {
+                        Associativity::Left => precedence + 1,
+                        Associativity::Right => precedence,
+                    },
+                ) {
+                    rhs.abandon(parser);
+                    return Err((infix, e));
+                }
+
+                lhs = infix.complete(parser, kind);
+            }
+            Some((token, _)) if POSTFIX_OPERATOR_EXPECTED_SET.contains(token.into_raw()) => {
+                // The operation is a postfix operation; check the precedence level
+                let precedence = postfix_precedence(token);
+                if precedence < min_precedence {
+                    break;
+                }
+
+                // Call the operation-specific parse function
+                let postfix = lhs.precede(parser);
+                let res = match token {
+                    Token::OpenParen => call_expr(parser, postfix),
+                    Token::OpenBracket => index_expr(parser, postfix),
+                    Token::Dot => access_expr(parser, postfix),
+                    _ => panic!("unexpected postfix operator"),
+                };
+
+                lhs = match res {
+                    Ok(marker) => marker,
+                    Err((postfix, e)) => {
+                        return Err((postfix, e));
+                    }
+                };
+            }
+            _ => break,
+        }
+    }
+
+    Ok(lhs)
+}
+
+/// Parses an atomic expression such as a literal.
+///
+/// Due to the WDL grammar having an ambiguity between parenthesized expressions
+/// and pair literals, this function also handles the former.
```

Comment:
```suggestion
/// Due to the WDL grammar having an ambiguity between parenthesized expressions
/// and pair literals, this function handles the former in addition to pair literals.
```
I initially misread this comment. It was grammatically correct, but I found it confusing. I think this is a bit better.